### PR TITLE
Configure node version management (nvm)

### PR DESCRIPTION
**Pull request by johnclary**
This PR adds support for [Node Version Manager](https://github.com/nvm-sh/nvm) to our Moped environments. 

## Associated issues
Resolves https://github.com/cityofaustin/atd-data-tech/issues/8309. 
See discussion in https://github.com/cityofaustin/atd-data-tech/issues/7819.

## Testing

This should be tested at **all** three of local, Netlify deploy preview, and moped-test.

**Steps to test:**
To test locally, follow the instructions in the [updated readme](https://github.com/cityofaustin/atd-moped/tree/jc-nvm-setup/moped-editor). I found the [zsh-nvm](https://github.com/lukechilds/zsh-nvm) install to be straightforward.

Going forward, our Netlify and Github Action builds will use the same Node.js runtime environment (and paired `npm` version) when building the Moped editor, according to the version defined in the `.nvmrc` file. I've pinned it to major release `16`. We could be more specific, but Node is [quite clear](https://nodejs.org/en/about/releases/) about their versioning system and support schedule and pinning to a major release seems sound.

This will have the immediate effect of bumping our `node` versions in both Netlify and moped-test. Netlify currently uses node `v12` and the moped-test github action uses an earlier version of `16`.

It is _upon us_ to adopt the habit of using `nvm` locally to ensure our build env stays in sync.

Other than testing locally, I'm not really sure you need to test anything on the preview branch or moped-test. The builds are successful, and you can verify that by checking the build logs and navigating to them.

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

**Comment by johnclary** [comment]
This preview [built](https://app.netlify.com/sites/atd-moped-main/deploys/62ab83bc1e860f000997c1ef) with `node v16.15.1` thanks to `.nvmrc`.

```
3:26:44 PM: Started restoring cached node version
3:26:45 PM: Finished restoring cached node version
3:26:45 PM: Attempting node version '16' from .nvmrc
3:26:46 PM: Downloading and installing node v16.15.1...
3:26:46 PM: Downloading https://nodejs.org/dist/v16.15.1/node-v16.15.1-linux-x64.tar.xz...
3:26:46 PM: Computing checksum with sha256sum
3:26:47 PM: Checksums matched!
3:26:49 PM: Now using node v16.15.1 (npm v8.11.0)
```

Compare to a normal netlify build, with node cached at `v12`:

```
1:14:41 PM: Started restoring cached node version
1:14:43 PM: Finished restoring cached node version
1:14:43 PM: v12.18.0 is already installed.
1:14:44 PM: Now using node v12.18.0 (npm v6.14.4)
```

File: .github/workflows/atd_moped_build_stage.yml [hunk 1]
```diff
@@ -15,16 +15,17 @@ jobs:
     name: Build Editor
     runs-on: ubuntu-latest
     steps:
+      # Get the code first
+      - name: "Checkout"
+        uses: actions/checkout@v2
```
Comment (johnclary) [inline .github/workflows/atd_moped_build_stage.yml:20]: the python step (below) does not need this step, but i learned the [hard way](https://stackoverflow.com/questions/64405836/github-actions-no-such-file-or-directory-on-any-run-step) that you need to `checkout` before you can access files in your repo 🤦

File: .github/workflows/atd_moped_build_stage.yml [hunk 2]
```diff
@@ -15,16 +15,17 @@ jobs:
...
       - uses: actions/setup-python@v2
         with:
           python-version: "3.8"
           architecture: "x64"
-      - uses: actions/setup-node@v2
+      # Install node according to our .nvmrc file
+      - uses: actions/setup-node@v3
```
Comment (johnclary) [inline .github/workflows/atd_moped_build_stage.yml:26]: bumping from `setup-node@v2` to `setup-node@v3`, which merely dropped support for pre-v16 versions of node.

**Comment by johnclary** [comment]
devs, no rush here. i'm going to branch from this one as i take on some other NPM-related tasks. thanks for your review!

**Review by frankhereford — APPROVED** [review]
👍 💯 Thanks for this - nvm does just what it says on the tin, and it's just what we need. Thanks for laying some more foundation for updating the app's js dependencies. 💪 

I will `nvm use` going forward - ✅

**Review by mddilley — APPROVED** [review]
This looks great! Local ✅ Deploy preview ✅ Moped-test ✅  And, TIL more about GH Actions. 👍 

This reminded me of a Vercel thing called Fast Node Manager https://fnm.vercel.app/, and it gave me a reason to try it out since it supports .nvmrc files too. Works just fine but I'll stick to nvm so we all use the same tool.

**Review by mateoclarke — APPROVED** [review]
Worked as expected for me on local.
🚢 💨✨

**Review by chiaberry — APPROVED** [review]

**Pull request merged**